Panic in OnAddOperation

`OnAddOperation` can panic [here](https://github.com/danielgtaylor/huma/blob/v2.12.0/transforms.go#L128) if one of the fields of a return type is an embedded struct (there might be a requirement that the embedded struct also has some methods on it)

Panics with

```
panic: reflect: embedded type with methods not implemented if type is not first field
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -18,4 +18,6 @@ jobs:
         with:
           version: v1.57.2
       - run: go test -coverprofile=coverage.txt -covermode=atomic ./...
-      - uses: codecov/codecov-action@v1
+      - uses: codecov/codecov-action@v4
+        with:
+          token: ${{ secrets.CODECOV_TOKEN }}

--- a/huma_test.go
+++ b/huma_test.go
@@ -1465,6 +1465,23 @@ func TestConvenienceMethods(t *testing.T) {
 	assert.Equal(t, "delete-things-by-thing-id", api.OpenAPI().Paths[path].Delete.OperationID)
 }
 
+type EmbeddedWithMethod struct{}
+
+func (e EmbeddedWithMethod) Method() {}
+
+func TestUnsupportedEmbeddedTypeWithMethods(t *testing.T) {
+	_, api := humatest.New(t, huma.DefaultConfig("Test API", "1.0.0"))
+
+	// Should not panic!
+	huma.Post(api, "/things", func(ctx context.Context, input *struct{}) (*struct {
+		Body struct {
+			EmbeddedWithMethod
+		}
+	}, error) {
+		return nil, nil
+	})
+}
+
 // func BenchmarkSecondDecode(b *testing.B) {
 // 	//nolint: musttag
 // 	type MediumSized struct {

--- a/transforms.go
+++ b/transforms.go
@@ -2,6 +2,7 @@ package huma
 
 import (
 	"bytes"
+	"fmt"
 	"path"
 	"reflect"
 )
@@ -125,13 +126,23 @@ func (t *SchemaLinkTransformer) OnAddOperation(oapi *OpenAPI, op *Operation) {
 				}
 			}
 
-			newType := reflect.StructOf(fields)
-			info := t.types[typ]
-			info.t = newType
-			info.fields = fieldIndexes
-			info.ref = extra.Schema
-			info.header = "<" + extra.Schema + ">; rel=\"describedBy\""
-			t.types[typ] = info
+			func() {
+				defer func() {
+					if r := recover(); r != nil {
+						// Catch some scenarios that just aren't supported in Go at the
+						// moment. Logs an error so people know what's going on.
+						// https://github.com/danielgtaylor/huma/issues/371
+						fmt.Println("Warning: unable to create schema link for type", typ, ":", r)
+					}
+				}()
+				newType := reflect.StructOf(fields)
+				info := t.types[typ]
+				info.t = newType
+				info.fields = fieldIndexes
+				info.ref = extra.Schema
+				info.header = "<" + extra.Schema + ">; rel=\"describedBy\""
+				t.types[typ] = info
+			}()
 		}
 	}
 }

